src/wiring_diagrams/CPortGraphs.jl

```diff
@@ -0,0 +1,194 @@
+module CPortGraphs
+using ...Theories
+using ...CategoricalAlgebra
+import ...CategoricalAlgebra.CSets: migrate!
+using ...Present
+using ...Graphs
+using ...Graphs.BasicGraphs
+import ...Graphs.BasicGraphs: TheoryGraph
+import ..DirectedWiringDiagrams: ocompose
+
+export ThCPortGraph, ThOpenCPortGraph, ThSymCPortGraph, ThOpenSymCPortGraph, ThBundledCPG,
+  CPortGraph, OpenCPortGraph, SymCPortGraph, OpenSymCPortGraph, BundledCPG
+
+
+@present ThCPortGraph(FreeSchema) begin
+    B::Ob
+    P::Ob
+    W::Ob
+
+    src::Hom(W,P)
+    tgt::Hom(W,P)
+    box::Hom(P,B)
+end
+
+@present ThOpenCPortGraph <: ThCPortGraph begin
+    OP::Ob
+    con::Hom(OP, P)
+end
+
+@present ThSymCPortGraph <: ThCPortGraph begin
+    inv::Hom(W, W)
+
+    compose(inv,inv) == id(W)
+    compose(inv,src) == tgt
+    compose(inv,tgt) == src
+end
+
+@present ThOpenSymCPortGraph <: ThSymCPortGraph begin
+    OP::Ob
+
+    con::Hom(OP, P)
+end
+
+const CPortGraph = ACSetType(ThCPortGraph)
+const OpenCPortGraph = ACSetType(ThOpenCPortGraph)
+const SymCPortGraph = ACSetType(ThSymCPortGraph)
+const OpenSymCPortGraph = ACSetType(ThOpenSymCPortGraph)
+const OSCPGraph = OpenSymCPortGraph
+
+function OpenCPortGraph(g::CPortGraph)
+    x = OpenCPortGraph()
+    copy_parts!(x, g, W=parts(g, :W), P=parts(g, :P), B=parts(g, :B))
+    return x
+end
+
+function migrate!(g::Graph, cpg::CPortGraph) 
+    migrate!(g, cpg,
```

Comment:
I should have raised this in response to #317, but looking at the code for `migrate!` I see that it assumes that `g` is empty. IMO, that's confusing but it's easy enough to fix.

No need to worry about this here. I will fix this in a followup PR.

Reply:
Extension implemented in #343.